### PR TITLE
More Efficient Ember Encoding

### PR DESCRIPTION
**Pull request by ChristopherDavenport**
- It was previously UTF-8, we've switched to ASCII to match the [gist](https://gist.github.com/rossabaker/31ea688c2337c40a0a0a7070cc8b4e1d), is that the correct encoding?

### CODE REVIEW

File: ember-core/src/main/scala/org/http4s/ember/core/Encoder.scala [hunk 1]
```diff
@@ -9,42 +9,75 @@ package org.http4s.ember.core
...
+      // Apply each header followed by a CRLF
+      resp.headers.foreach { h =>
+        stringBuilder
+          .append(h.renderString)
+          .append(CRLF)
+        ()
+      }
+      // Final CRLF terminates headers and signals body to follow.
+      stringBuilder.append(CRLF)
+      stringBuilder.toString.getBytes(StandardCharsets.US_ASCII)
```
Comment (rossabaker) [inline ember-core/src/main/scala/org/http4s/ember/core/Encoder.scala:38]: Note that this (and line 76) can potentially throw, if there was something naughty in the model.  Our model constructors guard against a lot of these, but they're not foolproof yet.

Comment (rossabaker) [reply ember-core/src/main/scala/org/http4s/ember/core/Encoder.scala:38]: On second thought, there is an old tradition of smuggling ISO-8859-1 through this, and it achieves that 1:1 mapping between bytes and chars as ASCII. Perhaps that is the better choice.

Comment (rossabaker) [reply ember-core/src/main/scala/org/http4s/ember/core/Encoder.scala:38]: The [RFC says](https://tools.ietf.org/html/rfc7230#section-3.2.4):

> Historically, HTTP has allowed field content with text in the
   ISO-8859-1 charset [ISO-8859-1], supporting other charsets only
   through use of [RFC2047] encoding.  In practice, most HTTP header
   field values use only a subset of the US-ASCII charset [USASCII].
   Newly defined header fields SHOULD limit their field values to
   US-ASCII octets.  A recipient SHOULD treat other octets in field
   content (obs-text) as opaque data.

**Comment by rossabaker** [comment]
Are we keeping this off 0.21 because it changes undefined (by us or spec) behavior?  It looks binary compatible, and I wouldn't necessarily mind backporting it.

**Comment by ChristopherDavenport** [comment]
Updated Charset

**Pull request merged**